src/pip/_internal/download.py

```diff
@@ -455,14 +455,27 @@ def get_file_content(url, comes_from=None, session=None):
 
 
 def is_url(name):
-    # type: (Union[str, Text]) -> bool
+    # type: (str) -> bool
     """Returns true if the name looks like a URL"""
     if ':' not in name:
         return False
-    scheme = name.split(':', 1)[0].lower()
+    scheme = split_scheme_from_url(name)[0]
     return scheme in ['http', 'https', 'file', 'ftp'] + vcs.all_schemes
 
 
+def split_scheme_from_url(url):
```

Comment:
I feel like generic parsing functions like this should go in `misc.py` with the other URL parsing functions. (Incidentally, I also think that `path_to_url()` and friends shouldn't be in `download.py` either.)

Reply:
@cjerdonek `url_to_path()` as well?

Reply:
Looking at `download.py`, these URL related functions should stay together, either at `misc.py` or at `download.py`. But `misc.py` looks really polluted to me. Maybe creating an `utils` package that contains a file for URL related functions would be a better idea?

Reply:
Yes, probably. :) But I don’t want to sidetrack this PR further. For functions other than new functions you’re adding here, it would need to be done as a separate PR. I’m also not sure what type of function you’ll wind up needing after your conversation with @uranusjr (I haven’t thought about it myself).

Reply:
Great, I would really like to tackle this up :)
I'll add this function to the `misc.py` then, if I still have to use it.